lang/cs/Org.Apache.REEF.Common/Runtime/Evaluator/Context/RootContextLauncher.cs

```diff
@@ -40,15 +41,16 @@ internal sealed class RootContextLauncher
         private ContextRuntime _rootContext = null;
 
         public RootContextLauncher(string id, IConfiguration contextConfiguration,
-            Optional<ServiceConfiguration> rootServiceConfig, Optional<TaskConfiguration> rootTaskConfig)
+            Optional<ServiceConfiguration> rootServiceConfig, Optional<IConfiguration> rootTaskConfig, IHeartBeatManager heartbeatManager)
         {
             Id = id;
             _rootContextConfiguration = contextConfiguration;
             _rootServiceInjector = InjectServices(rootServiceConfig);
+            _rootServiceInjector.BindVolatileInstance(GenericType<IHeartBeatManager>.Class, heartbeatManager);
```

Comment:
We are doing this because we read in `DriverConfiguration` in the Evaluator, which can cause a conflict at the context/task level by us forking injectors off of the base Evaluator injector. This issue is addressed by https://issues.apache.org/jira/browse/REEF-217, and I have added a `TODO` for this. Thanks!